assignments/prototypes.js

```diff
@@ -102,9 +126,110 @@
   console.log(archer.greet()); // Lilith offers a greeting in Elvish.
   console.log(mage.takeDamage()); // Bruce took damage.
   console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+ 
 
   // Stretch task: 
   // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
   // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villain and one a hero and fight it out with methods!
\ No newline at end of file
+  // * Create two new objects, one a villain and one a hero and fight it out with methods!
+// Hero  ==========================================================================================================
+  function Hero (gameData){
+    Humanoid.call(this, gameData);
+    this.attackPower = gameData.attackPower;    // Base damage done
+    this.critStrike = gameData.critStrike;      // Has a chance for an attack to do double damage
+    this.armor = gameData.armor;                // Flat % damage reduction
+  }
+  Hero.prototype = Object.create(Humanoid.prototype);
+  Hero.prototype.hDamageDone = function(){
+    // Random number is generated to determine whether the attack will trigger a critical strike for double damage.
+    let critRand = Math.random();
+    if (critRand > 1-(this.critStrike/10)){return this.attackPower*2}
+    else {return this.attackPower}
```

Comment:
This code is completely valid - however, personally I like to separate the if criteria into a variable and put the if statement on separate lines, like;

```const isCriticalStrike = critRand > 1 - (this.critStrike/10);```

Then you can do:

if (isCriticalStrike) {
    return this.attackPower * 2;
}
```

I just think it makes it easier to read and debug :)